bigquery-connector-common/pom.xml

```diff
@@ -27,6 +27,11 @@
       <groupId>com.google.api.grpc</groupId>
       <artifactId>proto-google-cloud-bigquerystorage-v1</artifactId>
     </dependency>
+    <dependency>
```

Comment:
lz4-java is provided by spark. If we can use that, then this dependencies can be removed. If we need version 1.8.0 then we need to shade it as well - the `net/jpountz/...` package as well as the `.so` library. Although, that the [native library loading](https://github.com/lz4/lz4-java/blob/master/src/java/net/jpountz/util/Native.java#L98) doesn't really support multiple versions, so perhaps using the spark provided version is the better option.

Reply:
When I remove org.lz4 lz4-java  from bigquery-connector-common/pom.xml, then I get this compilation in the bigquery-connector-common file

```
[ERROR] ~/spark-bigquery-connector/bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/DecompressReadRowsResponse.java:[8,23] package net.jpountz.lz4 does not exist
```

I think this is because although [lz4-java](https://mvnrepository.com/artifact/org.lz4/lz4-java) is included in [spark-core](https://mvnrepository.com/artifact/org.apache.spark/spark-core), that isn't included in the bigquery-connector-common/pom.xml.  
I do see org.apache.spark spark-sql_2.13 and org.apache.spark spark-mllib_2.13 in spark-bigquery-connector-common/pom.xml, but I want to use lz4 in bigquery-connector-common and not spark-bigquery-connector-common code. 
plus, neither [spark-sql_2.13](https://mvnrepository.com/artifact/org.apache.spark/spark-sql_2.13/3.5.0) nor [spark-mllib_2.13](https://mvnrepository.com/artifact/org.apache.spark/spark-mllib) include lz4-java. 
So it seems like currently, bigquery-connector-common doesn't contain a spark sql or spark ml lib dependency containing a lz4-java dependency.

Reply:
we decided to mark lz4-java as a provided dependency for bigquery-connector-common